api/dao/containerutil.py

```diff
@@ -0,0 +1,28 @@
+import bson.objectid
+from .. import config
+
+log = config.log
+
+
+def add_id_to_subject(subject, pid):
+    """
+    Add a mongo id field to given subject object (dict)
+
+    Use the same _id as other subjects in the session's project with the same code
+    If no _id is found, generate a new _id
+    """
+    result = None
+    if subject is None:
+        subject = {}
+    if subject.get('_id') is not None:
+        return subject
+    if subject.get('code') is not None:
```

Comment:
Suggest adding whitespace before this line to separate out the conditionals for readability